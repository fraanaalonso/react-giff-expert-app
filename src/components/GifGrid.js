import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GiffGridItem';


export const GifGrid = ( {category} ) => {

    const {data: images, loading} = useFetchGifs(category);


    return (
        <>
        <h3>{ category }</h3>

        { loading && <p className="animate__animated animate__flash">Loading...</p> }
        <div className='card-grid'>
                {
                    //I unstructured the array images
                    images.map( (img) => (
                    <GifGridItem
                        key = { img.id }
                        {...img}
                        
                    />
                    ))
                }
        
            </div> 
         </>
    )
}
