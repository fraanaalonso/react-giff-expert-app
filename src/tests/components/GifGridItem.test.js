import React from 'react'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GiffGridItem'


describe('Test GifGridItem component', () => {

    const url = 'https://localhost/algo.jpg';
    const title = 'Un titulo';

    const wrapper = shallow( <GifGridItem title={title} url={url} />)


    test('Must show a snapshot ', () => {
        const wrapper = shallow(<GifGridItem />)
        expect( wrapper ).toMatchSnapshot();

        
    })

    test('Must show title and url', () => {


        expect( wrapper ).toMatchSnapshot();

    })

    test('Must have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('Must have an image equal to url and props alt', () => {
        const img = wrapper.find('img');
        //console.log( img.prop('src') );
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    })
    
    test('Must have animated__fadein', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__fadeIn') ).not.toBe( false );
    })
    
    
})