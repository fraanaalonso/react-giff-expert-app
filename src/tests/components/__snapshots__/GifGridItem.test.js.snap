// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem component Must show a snapshot  1`] = `
<div
  className="card animate__fadeIn"
>
  <img />
  <p />
</div>
`;

exports[`Test GifGridItem component Must show title and url 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
